impl/csharp/Pika/Snowflakes/Snowflake.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace Pika.Snowflakes;
+
+public class Snowflake
+{
+    private readonly ulong _epoch;
+    private ulong _seq;
+    private long _lastSequenceExhaustion;
+
+    public Snowflake(ulong epoch, ulong nodeId)
+    {
+        _epoch = NormalizeEpoch(epoch);
+        NodeId = nodeId;
+    }
+
+    public ulong NodeId { get; }
+
+    public ulong Generate(SnowflakeGenOptions? options = null)
+    {
+        options ??= new SnowflakeGenOptions();
+        var timestamp = options.Timestamp ?? DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+        if (timestamp < _lastSequenceExhaustion)
+        {
+            throw new SequenceExhaustionError();
+        }
+
+        _lastSequenceExhaustion = timestamp;
+        if (timestamp == _lastSequenceExhaustion)
+        {
+            _seq++;
```

Comment:
Snowflakes should start at sequence 0, and this causes them to start at 1. The sequence should only be incremented if it reaches 4096. 

https://github.com/hopinc/pika/blob/main/impl/js/src/snowflake.ts#L82

Reply:
Nevermind, this still fits the spec!